### PR TITLE
MM-32773 - open file screen

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR covers the functionality for the open file screen from the import workspace project.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32773

### CODE REVIEW

**Comment by pablovelezvidal** [comment]
For testing purposes I used these two files:
[test-files.zip](https://github.com/mattermost/mattermost-webapp/files/6070709/test-files.zip)

File: components/admin_console/cloud_import/import_workspace.tsx [hunk 1]
```diff
@@ -58,14 +58,14 @@ const importList = [{
...
-    const [screenStep, setScreenStep] = useState(1);
+    const [percentage] = useState(7);
+    const [screenStep, setScreenStep] = useState(0);
     const [isImportInProgress] = useState(true);
     const [isNewWorkspace] = useState(true);
 
-    useInterval(() => {
-        setPercentage((prevPercentage) => prevPercentage + 1);
-    }, 1000, 99000);
+    // useInterval(() => {
```
Comment (AGMETEOR) [inline components/admin_console/cloud_import/import_workspace.tsx:66]: forgot commented code. this puts some ununsed import on line 15

**Review by AGMETEOR — APPROVED** [review]
hey Pablo thanks for the good work. lays good foundation for the upcoming migration functionalities. LGTM and just a small comment

File: components/admin_console/cloud_import/components/open_file.tsx [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+import {
+    CloudLinks,
+} from 'utils/constants';
+
+import OpenFileImg from '../open_file.svg';
+
+import './open_file.scss';
+import FilePreview from 'components/file_preview/file_preview';
+import AlertBanner from 'components/alert_banner';
```
Comment (marianunez) [inline components/admin_console/cloud_import/components/open_file.tsx:27]: To keep the imports organized, can we move these and the `UpgradeLink` above together with all the `components/...` imports above?

File: components/admin_console/cloud_import/components/open_file.tsx [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+    const [openFileState, setOpenFileState] = useState(FileState.INITIAL);
+
+    const handleDropCb = (e: any) => {
+        // TODO: TEMPORAL - validating and parsing a simple JSON file
+        const file = e.dataTransfer.files[0];
+        onFileLoaded(file);
+        setShowOverlay(false);
+    };
+
+    const handleDragOverCb = () => {
```
Comment (marianunez) [inline components/admin_console/cloud_import/components/open_file.tsx:60]: For readability, I would suggest `Cb` here on the naming. Doesn't provide any value and the `handle` naming convention may be explicit enough.
```suggestion
    const handleDragOver = () => {
```

File: components/admin_console/cloud_import/components/open_file.tsx [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+                fileStateCompletedSection(
+                    fileValues,
+                    chatService,
+                    removeFile,
+                    props.nextStepFunc,
+                    props.maxFreeUsers,
+                    isPaidTier,
+                    deactivateUsers,
+                    setDeactivateUsers)}
+        </CardContainer>
```
Comment (marianunez) [inline components/admin_console/cloud_import/components/open_file.tsx:149]: Would it be more idiomatic React to make these functions `openFileStateInitialSection` and `openFileStateInitialSection` a functional component with props?

File: components/admin_console/cloud_import/components/open_file.tsx [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+import FilePreview from 'components/file_preview/file_preview';
+import AlertBanner from 'components/alert_banner';
+
+export interface StepOneProps {
+    chatService: string;
+    nextStepFunc: () => void;
+    maxFreeUsers: number;
+}
+
+const StepOne: React.FC<StepOneProps> = (props: StepOneProps) => {
```
Comment (marianunez) [inline components/admin_console/cloud_import/components/open_file.tsx:35]: Should this be called `OpenFile` or `OpenImportFile` given the filename `StepOne` doesn't seem to provide enough context at this component level.

**Review by marianunez — CHANGES_REQUESTED** [review]
Great stuff with the drag and drop and optimizing the rendering while updating the status! 🎉

Some minor comments below mostly on code org a bit more the `openFile` component

**Comment by marianunez** [comment]
@pvev should we close this PR for now? We can maybe bring back this branch when we resume this work (not scheduled at this time)

**Pull request closed without merging**